### PR TITLE
Split up vrt id token validation and determining if its a valid vrt in the current VRT schema

### PR DESCRIPTION
**Pull request by maschwenk**
Base branch #11 

1. Added some caching (prepended the cache ivar names for best practice)
2. Split up the `valid?` method into:
    a. Token validation
    b. making sure it's a valid node W.R.T. the schema
3. Make sure lineages return `nil` when the vrt id is invalid rather than throwing an exception

The caching here is probably not so important given that most of the time we are creating entirely new `Vrt::Map` instances. Maybe there's a case to be made that the VRT singleton should have a memo map of all the `VRT::Map`'s for each version (or at least start pre-warmed with the current version)? Either way I'd address separately.

### CODE REVIEW

File: lib/vrt/map.rb [hunk 1]
```diff
@@ -54,7 +58,6 @@ def construct_lineage(string, max_depth)
 
     def walk_node_tree(string, max_depth: 'variant')
       id_tokens = string.split('.').map(&:to_sym)
-      return nil if id_tokens.size > 3
```
Comment (maschwenk) [inline lib/vrt/map.rb]: This was basically a double verification of the id, and is definetly OOS of what this function is doing, that's why I removed it

File: lib/vrt/map.rb [hunk 2]
```diff
@@ -43,9 +40,16 @@ def categories
 
     private
 
+    def valid_identifier?(vrt_id)
+      # At least one string of lowercase or _, plus up to 2 more with stops
+      @_valid_identifiers[vrt_id] ||= vrt_id =~ /other|\A[a-z_]+(\.[a-z_]+){0,2}\z/
```
Comment (maschwenk) [inline lib/vrt/map.rb]: Just figured I'd configure the `other` into the regex, lmk if you don't like that

File: lib/vrt/map.rb (hunk 1 above)
Comment (arcwhite) [reply lib/vrt/map.rb]: Yep, valid!

File: lib/vrt/map.rb (hunk 2 above)
Comment (arcwhite) [reply lib/vrt/map.rb]: Nah, I dig it

Comment (adamrdavid) [reply lib/vrt/map.rb]: I personally find regex super confusing and avert my eyes at all costs.   not totally against it though.

File: lib/vrt/map.rb [hunk 3]
```diff
@@ -11,25 +11,22 @@ class Map
...
 
-    def valid?(node)
-      return true if node == 'other'
-      # At least one string of lowercase or _, plus up to 2 more with stops
-      return false unless node =~ /\A[a-z_]+(\.[a-z_]+){0,2}\z/
-      found_node = find_node(node)
-      return false unless found_node
-      true
+    def valid?(vrt_id)
+      @_valid_vrt_ids[vrt_id] ||= valid_identifier?(vrt_id) && find_node(vrt_id)
```
Comment (arcwhite) [inline lib/vrt/map.rb]: This is returning nil, or a found node. I think by convention we should be coercing the result to a boolean

Comment (adamrdavid) [reply lib/vrt/map.rb]: doesn't the && coerce it into a bool?

File: lib/vrt/map.rb (hunk 2 above)
Comment (adamrdavid) [reply lib/vrt/map.rb]: its is nice to have it all handled by one case.

Comment (maschwenk) [reply lib/vrt/map.rb]: I agree in general, but at least its just a single literal of `other` and then regex horror show

File: lib/vrt/map.rb (hunk 3 above)
Comment (maschwenk) [reply lib/vrt/map.rb]: no, you're right, when I was looking at this, rubocop was very offended by `!!` but it can be acheived in other ways

Comment (adamrdavid) [reply lib/vrt/map.rb]: my b, I'm still learning ruby lol.

Comment (maschwenk) [reply lib/vrt/map.rb]: lol you don't know how many times I run that locally in both javascript and ruby and question my existence

**Pull request merged**